src/inference/enumerate.ad.js

```diff
@@ -67,21 +93,33 @@ module.exports = function(env) {
     // Find support of this distribution:
     if (dist.isContinuous || !dist.support) {
       console.error(dist);
-      throw new Error('Enumerate can only be used with distributions that have finite support.');
+      return 'Enumerate can only be used with distributions that have finite support.';
     }
     var supp = dist.support();
 
     // Check that support is non-empty
     if (supp.length === 0) {
       console.error(dist);
-      throw new Error('Enumerate encountered a distribution with empty support!');
+      return 'Enumerate encountered a distribution with empty support.';
     }
-
     return supp;
   };
 
   Enumerate.prototype.sample = function(store, k, a, dist) {
     var support = getSupport(dist);
+    if (_.isString(support)) {
```

Comment:
We need to pass a function that represents the computation that happens after `getSupport` -- this is the continuation. So the body of the `sample` method will go from something like this:
```js
var supp = getSupport(dist);
// rest of body
```
to this:
```js
return getSupport(dist, function(support) {
  // rest of body
});
```